frontend/src/store/objectStore.ts

```diff
@@ -39,8 +36,20 @@ export const useObjectStore = defineStore('objectStore', () => {
 
   async function listObjects(params = {}) {
     objectList.value = [];
-    const response = await objectService.listObjects(params);
-    objectList.value = response.data;
+    const objects = (await objectService.listObjects(params)).data;
+
+    await Promise.all(
+      objects.map(async (obj: any) => {
```

Comment:
i expect you know what youre going to do here following the proposed process of factoring in permissions.
i guess it makes sense to make separate service calls and combine the data here.
the calls to listObjects and getMetadata can now accept a list of object id's

Reply:
Will look into cleaning this up as part of https://github.com/bcgov/bcbox/tree/feature/object-list-perm-check